src/uint.rs

```diff
@@ -1070,28 +1212,46 @@ where
 pub trait GetBit<I> {
     #[allow(missing_docs)]
     type Output;
+
+    #[doc(hidden)]
+    fn get_bit<IM: InternalMarker>(&self, &I) -> Self::Output;
 }
 
 #[allow(missing_docs)]
 pub type GetBitOut<N, I> = <N as GetBit<I>>::Output;
 
 // Base case
-impl<Un, Bn> GetBit<U0> for UInt<Un, Bn> {
+impl<Un, Bn> GetBit<U0> for UInt<Un, Bn>
+where
+    Bn: Copy
```

Comment:
The `Bn: Copy` bound is needed so we can produce an instance of the least significant bit from a reference to a `UInt`.

Reply:
I have no objection to this, but couldn't we alternatively make `get_bit` take things by value?

Reply:
We could! We'd just need to add an extra `Copy` bound to the `PartialDiv*` implementations.